src/proxy/inbound.rs

```diff
@@ -98,13 +103,21 @@ impl Inbound {
                 .http2_initial_stream_window_size(self.cfg.window_size)
                 .http2_initial_connection_window_size(self.cfg.connection_window_size)
                 .http2_max_frame_size(self.cfg.frame_size)
-                .serve(service);
+                .serve(service)
+                .with_graceful_shutdown(async {
+                    let shutdown = self.drain.signaled().await;
+                    // TODO: add some more graceful shutdown way
+                    time::sleep(time::Duration::from_secs(3)).await;
```

Comment:
This needs to be configurable or our tests will take forever

Reply:
yes, maybe infer from the existing one. i will remove it ATM